src/pathplanning.org

```diff
@@ -0,0 +1,130 @@
+6. Smooth out path with overlaid Bézier curve
+
+* How to construct a Bézier curve
+
+** Linear curve
+- Bézier curve B(t)
+- The t in the function for a linear Bézier curve can be thought of as describing how far B(t) is from P0 to P1
+- For example, when t=0.25, B(t) is one quarter of the way from point P0 to P1. As t varies from 0 to 1, B(t) describes a straight line from P0 to P1
+[[file:https://upload.wikimedia.org/wikipedia/commons/0/00/B%C3%A9zier_1_big.gif]]
+
+** Quadratic curve
+- For quadratic Bézier curves one can construct intermediate points Q0 and Q1 such that as t increases from 0 to 1
+  - Point Q0(t) varies from P0 to P1 and describes a linear Bézier curve.
+  - Point Q1(t) varies from P1 to P2 and describes a linear Bézier curve.
+  - Point B(t) is interpolated linearly between Q0(t) to Q1(t) and describes a quadratic Bézier curve.
+[[file:https://upload.wikimedia.org/wikipedia/commons/3/3d/B%C3%A9zier_2_big.gif]]
+# This can be applied to higher and higher orders of magnitude.
+
+* Sounds really inefficient
+- Why waste computation time branching out in random directions?
+- What advantages could there be in random branching?
+- Why not use a less computationally intense algorithm like A*?
+# What a good question
+
+** Advantages of RRT
+- Specialized for continuous configuration spaces
+- Fast
+- Can handle kinodynamic constraints
+- Algorithm can be modified for various needs and preferences
+
+* RoboCup Modifications
+
+** Biases
+- Goal Bias
+  - Random branching has tendency to branch directly towards goal instead
+- Waypoint Bias
+  - Random branching has tendency to branch towards Bézier curve waypoints of previous paths
+- Goal Bias + Waypoint Bias must sum to at most 1.0
+
+** Adaptive Stepsize Control
+- Stepsize now dynamically changes based on whether there are obstacles nearby
+- Requires extra computation time to locate nearby obstacles
+- Having larger stepsizes when possible reduces total iteration count, which reduces computation time
+- Obstacle-light environments benefit the most from this enhancement
+
+** k-d Trees
+- Finding the node in the RRT to extend from is computationally expensive
+- Binary tree that attempts to maintain balance
+- Partitions the state space evenly based on the number of nodes in each partition
+- The branch that each node is stored under is determined by comparing that node to the median of the node in that subtree
+
+* STOx's Planner
+# http://fei.edu.br/rcs/2014/RegularPapers/robocupsymposium2014_submission_55.pdf
+# Columbian RoboCup team, developed this nameless algorithm in 2013
+- Generate a straight line from the start state to end state
+- As long as the path intersects an obstacle:
+  - Generate a subgoal state next to the obstacle
+  - Now divided into two smaller subproblems
+  - Recurse!
+
+* STOx's Planner
+[[file:http:/i.imgur.com/Ea040em.png]]
+
+* STOx's Planner
+- Very fast when obstacle count is low
+- Not very flexible
+
+* A*
```

Comment:
I'm not sure about this, but this presentation seems slightly backwards. 

You reference A* in the RRT section, then talk about Dijkstras in the A* section, and then talk about Dijkstras. 

I think it might be a bit better to move the Dijsktras stuff first, then A*, then RRT (or remove the references to A* in the RRT section)